src/app/store/account.slice.ts

```diff
@@ -64,37 +77,51 @@ export const getAccounts = createSelector(
  */
 export function* accountsSaga() {
   yield all([
-    takeLatest(fetchAccount.type, fetchAccountWorker),
+    takeLatest(fetchAccounts.type, fetchAccountsWorker),
     takeLatest(removeAccount.type, removeAccountWorker)
   ]);
 }
 
-export function* fetchAccountWorker({
-  payload: wallet
-}: PayloadAction<SerializedWallet & { persistent: boolean }>) {
-  const address: TAddress = yield call(ipcBridgeRenderer.crypto.invoke, {
-    type: CryptoRequestType.GET_ADDRESS,
-    wallet
-  });
-
-  const uuid = generateDeterministicAddressUUID(address);
-
-  if (wallet.persistent) {
-    yield call(ipcBridgeRenderer.db.invoke, {
-      type: DBRequestType.SAVE_ACCOUNT_SECRETS,
-      wallet
-    });
+export function* fetchAccountsWorker({
+  payload: wallets
+}: PayloadAction<(SerializedWallet & { persistent: boolean })[]>) {
+  const accounts: IAccount[] = yield select(getAccounts);
+
+  try {
+    for (let i = 0; i < wallets.length; i++) {
+      const wallet = wallets[i];
```

Comment:
This can be simplified a bit
```suggestion
    for (const wallet of wallets) {
```